core/contractsdk/cpp/example/trustops/src/trust_counter.cc

```diff
@@ -42,79 +42,151 @@ DEFINE_METHOD(Counter, store) {
 }
 
 // add adds two encrypted numbers
-// input is in format {"l":"key_l", "r":"key_r", "o":"key_output"}
+// input format: {"l":"key_l", "r":"key_r", "commitment": c1,
+//                "commitment2": c2}
 DEFINE_METHOD(Counter, add) {
   xchain::Context *ctx = self.context();
   TrustOperators to(ctx, 0);
+  std::string value;
 
-  std::map<std::string, std::string> args_map;
-  for (auto it = ctx->args().begin(); it != ctx->args().end(); ++it) {
-    std::string value;
-    // get each encrypted value
-    if (it->first != "o") {
-      ctx->get_object(it->second, &value);
-      args_map[it->first] = value;
-    } else {
-      args_map[it->first] = it->second;
-    }
+  TrustOperators::Operand left_op, right_op;
+  // get left operand
+  if (ctx->arg("l").empty() || ctx->arg("commitment").empty()) {
+      ctx->error("missing left operand parameter ");
+      return;
+  }
+  if (!ctx->get_object(ctx->arg("l"), &value)) {
+      ctx->error("get left operand error ");
+      return;
+  }
+  left_op.cipher = value;
+  left_op.commitment = ctx->arg("commitment");
+  // get right operand
+  if (ctx->arg("r").empty() || ctx->arg("commitment2").empty()) {
```

Comment:
Suggest check params at the beginning of the function to avoid extra costs.

Reply:
> Suggest check params at the beginning of the function to avoid extra costs.

already fixed